sdl-core/src/main/scala/io/smartdatalake/workflow/ActionDAGRun.scala

```diff
@@ -208,7 +208,10 @@ private[smartdatalake] case class ActionDAGRun(dag: DAG[Action], executionId: SD
    */
   def getRuntimeInfos: Map[ActionId, RuntimeInfo] = {
     dag.getNodes.map { a =>
-      (a.id, a.getRuntimeInfo(Some(executionId)).getOrElse(RuntimeInfo(executionId, RuntimeEventState.PENDING)))
+      var runtimeInfo = a.getRuntimeInfo(Some(executionId)).getOrElse(RuntimeInfo(executionId, RuntimeEventState.PENDING))
+      // overwrite DataObjectsState for skipped actions to keep previous state.
+      if (runtimeInfo.state == RuntimeEventState.SKIPPED) runtimeInfo = runtimeInfo.copy(dataObjectsState = initialDataObjectsState)
```

Comment:
why do you need to copy it?

Reply:
Well, i want to update a field of the case class. How would you do this?

Reply:
ah of course, I forgot about that scala pattern